### PR TITLE
Fixed background state

### PR DESCRIPTION
**Pull request by mcartwright**
Added start and stop indices of the noise source files to background state. This is in response to Issue #39

### CODE REVIEW

**Comment by bmcfee** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/bmcfee/muda/40#-KW3jqKNQd_jeuExi07S:-KW3jqKNQd_jeuExi07T:b7cwupx) (no related file):*
Looks good so far.  Can you add a test case to verify that the state variables are correct?

---

*[muda/deformers/background.py, line 42 at r1](https://reviewable.io:443/reviews/bmcfee/muda/40#-KW3iu9Q2amobk2wnjTM:-KW3iu9Q2amobk2wnjTN:b-465pej) ([raw file](https://github.com/bmcfee/muda/blob/8b6b2b855d8a7f650145a57ae134d9fa1bac184c/muda/deformers/background.py#L42)):*
> ```Python
>     with psf.SoundFile(str(filename), mode='r') as soundf:
> 
>         n_target = int(np.ceil(n_samples * soundf.samplerate / sr))
> ```

You might want to explicitly `float(sr)` in the div here so that we can be sure it's doing floating division in python 2.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bmcfee/muda/40)*

**Comment by bmcfee** [comment]
Pinging back on this -- looks like we have enough issues to push out a point release in the immediate future.  Any interest in finishing this one up so we can include it?

File: muda/deformers/background.py [hunk 1]
```diff
@@ -29,13 +29,55 @@ def sample_clip(filename, n_samples, sr, mono=True):
...
+    -------
+    start : int
+        The sample index from `filename` at which the audio fragment starts
+    stop : int
+        The sample index from `filename` at which the audio fragment stops (e.g. y = audio[start:stop])
+    '''
+
+    with psf.SoundFile(str(filename), mode='r') as soundf:
+
+        n_target = int(np.ceil(n_samples * soundf.samplerate / sr))
```
Comment (bmcfee) [inline muda/deformers/background.py]: `sr` should be explicitly cast here as `float(sr)` to ensure floating-point division in legacy python

**Comment by mcartwright** [comment]
Hey, yeah, I will make sure I get to this soon. It won't happen today, but probably tomorrow.

**Comment by mcartwright** [comment]
Yo, added some tests.

**Comment by bmcfee** [comment]
Thanks!

This looks like you merged master though. Could you rebase instead to keep the commit log clean?

**Comment by mcartwright** [comment]
Cleaned up history

**Comment by bmcfee** [comment]
Thanks!

**Pull request merged**